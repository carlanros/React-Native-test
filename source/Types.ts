
import * as login from './components/Login/types';

/**
 * All possible objects in Redux store
 *
 * @export
 * @interface IApplicationState
 */
export interface IApplicationState {
    login?: login.ILoginState;
}

/**
 * Describes possible props for a components.
 *
 * All actions and state objects might not be available.
 *
 * @export
 * @interface IProps
 */
export interface IProps extends IApplicationState {
    dispatch?: any;
    FacebookLogin?: login.FacebookLogin;
}