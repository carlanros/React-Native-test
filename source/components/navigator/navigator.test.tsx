/// <reference types="jest"/>
import React from 'react';
import { shallow } from 'enzyme';

import Navigator from './index';
import * as types from '../../Types';

jest.unmock('react-native');

function setup() {

  const enzymeWrapper = shallow(<Navigator /> );

  return {
    enzymeWrapper,
  };
}

describe('components', () => {
  /**
   * Test component rendering. Properties of children might be tested by importing their type:
   *   import { Text } from 'react-native';
   *    const busyProps = enzymeWrapper.find(Text).props();
   *    expect(busyProps.text).toBe('test);
   */
  describe('Navigator', () => {
    it('should render self and subcomponents', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});