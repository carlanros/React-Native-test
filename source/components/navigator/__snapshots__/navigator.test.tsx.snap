// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Navigator should render self and subcomponents 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppNavigator />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <View
        hideNavBar={true}
        panHandlers={null}
      >
        <View>
          <UNDEFINED
            activeBackgroundColor="blue"
            activeTintColor="yellow"
            inactiveBackgroundColor="black"
            inactiveTintColor="grey"
            lazy={true}
            tabBarPosition="bottom"
          >
            <View
              back={false}
              component={undefined}
              hideNavBar={true}
              icon={[Function]}
              tabBarLabel="Start"
              tintColor="white"
            />
            <View
              back={false}
              component={undefined}
              hideNavBar={true}
              icon={[Function]}
              tabBarLabel="Login 2"
              tintColor="white"
            />
          </UNDEFINED>
          <View
            back={false}
            component={undefined}
            hideNavBar={true}
          />
        </View>
      </View>,
      "uriPrefix": "myapp://",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": "root",
      "nodeType": "host",
      "props": Object {
        "children": <View>
          <UNDEFINED
            activeBackgroundColor="blue"
            activeTintColor="yellow"
            inactiveBackgroundColor="black"
            inactiveTintColor="grey"
            lazy={true}
            tabBarPosition="bottom"
          >
            <View
              back={false}
              component={undefined}
              hideNavBar={true}
              icon={[Function]}
              tabBarLabel="Start"
              tintColor="white"
            />
            <View
              back={false}
              component={undefined}
              hideNavBar={true}
              icon={[Function]}
              tabBarLabel="Login 2"
              tintColor="white"
            />
          </UNDEFINED>
          <View
            back={false}
            component={undefined}
            hideNavBar={true}
          />
        </View>,
        "hideNavBar": true,
        "panHandlers": null,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": "modal",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <UNDEFINED
              activeBackgroundColor="blue"
              activeTintColor="yellow"
              inactiveBackgroundColor="black"
              inactiveTintColor="grey"
              lazy={true}
              tabBarPosition="bottom"
            >
              <View
                back={false}
                component={undefined}
                hideNavBar={true}
                icon={[Function]}
                tabBarLabel="Start"
                tintColor="white"
              />
              <View
                back={false}
                component={undefined}
                hideNavBar={true}
                icon={[Function]}
                tabBarLabel="Login 2"
                tintColor="white"
              />
            </UNDEFINED>,
            <View
              back={false}
              component={undefined}
              hideNavBar={true}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "tabMain",
            "nodeType": "function",
            "props": Object {
              "activeBackgroundColor": "blue",
              "activeTintColor": "yellow",
              "children": Array [
                <View
                  back={false}
                  component={undefined}
                  hideNavBar={true}
                  icon={[Function]}
                  tabBarLabel="Start"
                  tintColor="white"
                />,
                <View
                  back={false}
                  component={undefined}
                  hideNavBar={true}
                  icon={[Function]}
                  tabBarLabel="Login 2"
                  tintColor="white"
                />,
              ],
              "inactiveBackgroundColor": "black",
              "inactiveTintColor": "grey",
              "lazy": true,
              "tabBarPosition": "bottom",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "main",
                "nodeType": "host",
                "props": Object {
                  "back": false,
                  "component": undefined,
                  "hideNavBar": true,
                  "icon": [Function],
                  "tabBarLabel": "Start",
                  "tintColor": "white",
                },
                "ref": null,
                "rendered": null,
                "type": "View",
              },
              Object {
                "instance": null,
                "key": "main2",
                "nodeType": "host",
                "props": Object {
                  "back": false,
                  "component": undefined,
                  "hideNavBar": true,
                  "icon": [Function],
                  "tabBarLabel": "Login 2",
                  "tintColor": "white",
                },
                "ref": null,
                "rendered": null,
                "type": "View",
              },
            ],
            "type": undefined,
          },
          Object {
            "instance": null,
            "key": "login",
            "nodeType": "host",
            "props": Object {
              "back": false,
              "component": undefined,
              "hideNavBar": true,
            },
            "ref": null,
            "rendered": null,
            "type": "View",
          },
        ],
        "type": "View",
      },
      "type": "View",
    },
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <View
          hideNavBar={true}
          panHandlers={null}
        >
          <View>
            <UNDEFINED
              activeBackgroundColor="blue"
              activeTintColor="yellow"
              inactiveBackgroundColor="black"
              inactiveTintColor="grey"
              lazy={true}
              tabBarPosition="bottom"
            >
              <View
                back={false}
                component={undefined}
                hideNavBar={true}
                icon={[Function]}
                tabBarLabel="Start"
                tintColor="white"
              />
              <View
                back={false}
                component={undefined}
                hideNavBar={true}
                icon={[Function]}
                tabBarLabel="Login 2"
                tintColor="white"
              />
            </UNDEFINED>
            <View
              back={false}
              component={undefined}
              hideNavBar={true}
            />
          </View>
        </View>,
        "uriPrefix": "myapp://",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": "root",
        "nodeType": "host",
        "props": Object {
          "children": <View>
            <UNDEFINED
              activeBackgroundColor="blue"
              activeTintColor="yellow"
              inactiveBackgroundColor="black"
              inactiveTintColor="grey"
              lazy={true}
              tabBarPosition="bottom"
            >
              <View
                back={false}
                component={undefined}
                hideNavBar={true}
                icon={[Function]}
                tabBarLabel="Start"
                tintColor="white"
              />
              <View
                back={false}
                component={undefined}
                hideNavBar={true}
                icon={[Function]}
                tabBarLabel="Login 2"
                tintColor="white"
              />
            </UNDEFINED>
            <View
              back={false}
              component={undefined}
              hideNavBar={true}
            />
          </View>,
          "hideNavBar": true,
          "panHandlers": null,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "modal",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <UNDEFINED
                activeBackgroundColor="blue"
                activeTintColor="yellow"
                inactiveBackgroundColor="black"
                inactiveTintColor="grey"
                lazy={true}
                tabBarPosition="bottom"
              >
                <View
                  back={false}
                  component={undefined}
                  hideNavBar={true}
                  icon={[Function]}
                  tabBarLabel="Start"
                  tintColor="white"
                />
                <View
                  back={false}
                  component={undefined}
                  hideNavBar={true}
                  icon={[Function]}
                  tabBarLabel="Login 2"
                  tintColor="white"
                />
              </UNDEFINED>,
              <View
                back={false}
                component={undefined}
                hideNavBar={true}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "tabMain",
              "nodeType": "function",
              "props": Object {
                "activeBackgroundColor": "blue",
                "activeTintColor": "yellow",
                "children": Array [
                  <View
                    back={false}
                    component={undefined}
                    hideNavBar={true}
                    icon={[Function]}
                    tabBarLabel="Start"
                    tintColor="white"
                  />,
                  <View
                    back={false}
                    component={undefined}
                    hideNavBar={true}
                    icon={[Function]}
                    tabBarLabel="Login 2"
                    tintColor="white"
                  />,
                ],
                "inactiveBackgroundColor": "black",
                "inactiveTintColor": "grey",
                "lazy": true,
                "tabBarPosition": "bottom",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "main",
                  "nodeType": "host",
                  "props": Object {
                    "back": false,
                    "component": undefined,
                    "hideNavBar": true,
                    "icon": [Function],
                    "tabBarLabel": "Start",
                    "tintColor": "white",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "View",
                },
                Object {
                  "instance": null,
                  "key": "main2",
                  "nodeType": "host",
                  "props": Object {
                    "back": false,
                    "component": undefined,
                    "hideNavBar": true,
                    "icon": [Function],
                    "tabBarLabel": "Login 2",
                    "tintColor": "white",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "View",
                },
              ],
              "type": undefined,
            },
            Object {
              "instance": null,
              "key": "login",
              "nodeType": "host",
              "props": Object {
                "back": false,
                "component": undefined,
                "hideNavBar": true,
              },
              "ref": null,
              "rendered": null,
              "type": "View",
            },
          ],
          "type": "View",
        },
        "type": "View",
      },
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
